README.asciidoc

```diff
@@ -635,8 +635,8 @@ use `n` followed by `,`, in order to remove a selection, use `<a-,>`.
 contains a match for this regex. Using `<a-K>` you can keep the selections
 not containing a match.
 
-`C` copies the current selection to the next line (or lines if a count is given)
-`<a-C>` does the same to previous lines.
+`C` duplicates selections on the lines that follow them.
+`<a-C>` does the same but on the preceding lines.
```

Comment:
nice, I never realized it worked for multiple selections on the same line.
Actually it's even more complicated than that. Looking at the source code it's something like roughly

```
`C`: for each selection that spans n lines, consider the following chunks of n lines each. Add a new selection on the first *count* chunks where the first/last line of the chunk are long enough to accomodate the column of the selection start/end
`<a-C>`: the same, but on preceding lines
```

which actually doesn't seem right. Below patch has the behavior I expect, what do you think?

It simplifies the description, which is a good sign

```
`C`: for each selection, add *count* selections with the same height starting at the next line where the tentative start/end lines are long enough to accomodate the column of the selection start/end.
`<a-C>`: the same, but on preceding lines
```

whether it's a good idea to go into such detail in the README is another question.

```patch
Author: Johannes Altmanninger <aclopte@gmail.com>
Date:   Sun Oct 9 03:29:17 2022 +0200

    Allow multiline C/<a-C> to match at offsets that are not multiples of the selection height
    
    Select the first <> pair in below file and press C. Nothing happens
    because Kakoune only tries create selections at lines that are
    multiples of 3 (the selection height). Fix this by considering every
    line, as long as it doesn't overlap with another selection we create.
    
             <
            A
            >
    
             <
            B
            >

diff --git a/src/normal.cc b/src/normal.cc
index 1608e77e4..93f3cce99 100644
--- a/src/normal.cc
+++ b/src/normal.cc
@@ -1456,10 +1456,9 @@ void copy_selections_on_next_lines(Context& context, NormalParams params)
         const LineCount height = std::max(anchor.line, cursor.line) - std::min(anchor.line, cursor.line) + 1;
         const size_t max_lines = std::max(params.count, 1);
 
+        LineCount offset = direction * height;
         for (size_t i = 0, nb_sels = 0; nb_sels < max_lines; ++i)
         {
-            LineCount offset = direction * (i + 1) * height;
-
             const LineCount anchor_line = anchor.line + offset;
             const LineCount cursor_line = cursor.line + offset;
 
@@ -1478,8 +1477,10 @@ void copy_selections_on_next_lines(Context& context, NormalParams params)
                 result.emplace_back(BufferCoord{anchor_line, anchor_byte},
                                     BufferCoordAndTarget{cursor_line, cursor_byte, cursor.target});
 
+                offset += direction * height;
                 nb_sels++;
-            }
+            } else
+                offset += direction;
         }
     }
     selections.set(std::move(result), main_index);
```

Reply:
In the old behavior `CC` does the same as `2C`, I'm not sure if this is true for this patch, however probably it would be easy to fix.

Reply:
> In the old behavior `CC` does the same as `2C`,

should still be the case with this patch